hooks/mari/collector.py

```diff
@@ -246,6 +246,24 @@ def collect_texture_elements(self, settings, parent_item):
                     self.logger.info("Collected item: %s" % layer_item.name)
                     items.append(layer_item)
 
+                    # for each layer, add associated mipmap item
+                    item_name = "%s, %s (%s mipmap)" % (geo.name(), channel.name(), layer_name)
+                    mipmap_item = self._add_item(settings,
+                                                layer_item,
+                                                item_name,
+                                                "mari.mipmap",
+                                                parent_item.context,
+                                                layer_properties)
+
+                    mipmap_item.set_thumbnail_from_path(thumbnail)
+                    mipmap_item.thumbnail_enabled = True
+
+                    # TODO: required?
+                    mipmap_item.properties.fields = self._resolve_item_fields(settings, mipmap_item)
```

Comment:
Shouldn't be. `self.on_context_changed()` gets called at `Item` initialization, which calls `self._resolve_item_fields()`

Reply:
Oh yes, I meant to get back and check this, but overlooked it.